### PR TITLE
On-board with using YAML-based release pipelines

### PR DESCRIPTION
**Pull request by JohnTortugo**
Relates to: https://github.com/dotnet/arcade/issues/3016

### CODE REVIEW

**Comment by JohnTortugo** [comment]
The diff is "big" but the actual new lines were just 5 for adding/converting jobs to stages. I'm running some private tests again with this before merging.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -11,213 +11,219 @@ resources:
...
-        - Asset_Registry_Publish
-      steps:
-        - checkout: self
-          clean: true
-        - powershell: eng/validation/update-channel.ps1
-            -maestroEndpoint https://maestro-prod.westus2.cloudapp.azure.com
-            -barToken $(MaestroAccessToken)
-          displayName: Move build to 'Latest' channel 
+- ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+  - template: eng\common\templates\post-build\release.yml
```
Comment (riarenas) [inline azure-pipelines.yml]: Isn't this called post-build.yml now?

**Comment by JohnTortugo** [comment]
Test build is here: https://dnceng.visualstudio.com/internal/_build/results?buildId=225864&view=results

The failure on `darc gather-drop` is due to a fixed bug in Tasks.Feed. The SDK used in the test build didn't use the fixed Tasks.Feed.

**Comment by JohnTortugo** [comment]
Here is a test build with all jobs passing: https://dnceng.visualstudio.com/internal/_build/results?buildId=226069&view=results

**Comment by JohnTortugo** [comment]
Can someone PTAL? I'd like to merge this today and if everything works fine onboard Arcade tomorrow.

**Comment by riarenas** [comment]
LGTM. We might want to see if we can change a bit how we promote the Arcade build to the latest channel (I think we'll have all the mechanisms in place for this, just needs to be done once we are ready for it)

**Comment by riarenas** [comment]
LGTM. We might want to see if we can change a bit how we promote the Arcade build to the latest channel (I think we'll have all the mechanisms in place for this, just needs to be done once we are ready for it)

**Comment by JohnTortugo** [comment]
Thanks Ricardo

**Pull request merged**